### ISSUE TITLE
Orphan option seems to break existing workflow

### ISSUE THREAD

**Issue by morberg** [stefanzweifel/git-auto-commit-action#97]
**Version of the Action**
`v4.5`

**Describe the bug**
I've been running this action for three months without problems. In version 4.5 something has changed in the action behavior and it no longer works as before.

If this is intentional perhaps version 4.5 should be version 5 instead if there are breaking changes?

**Output from workflow**
```bash
Run stefanzweifel/git-auto-commit-action@v4
  with:
    commit_message: Scheduled daily update
    file_pattern: .
    repository: .
    commit_user_name: GitHub Actions
    commit_user_email: actions@github.com
    commit_author: morberg <morberg@users.noreply.github.com>
    skip_dirty_check: false
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.8.5/x64
Started: bash /home/runner/work/_actions/stefanzweifel/git-auto-commit-action/v4/entrypoint.sh
+ _main
INPUT_REPOSITORY value: .
+ _switch_to_repository
+ echo 'INPUT_REPOSITORY value: .'
+ cd .
+ _git_is_dirty
++ git status -s -- .
+ '[' -n ' M odometer.csv
 M volvo.html' ']'
+ echo '::set-output name=changes_detected::true'
+ _switch_to_branch
INPUT_BRANCH value: 
+ echo 'INPUT_BRANCH value: '
+ git show-ref --verify --quiet refs/heads/
+ git checkout --orphan
error: option `orphan' requires a value
```

**Used Workflow**

```yaml
# Paste the Workflow you've used here
name: Update data and generate volvo.html

on:
  schedule:
    # Run nightly at 02:00 Swedish time, UTC is -2h from Sweden
    - cron: "00 00 * * *"
    
jobs:
  build:
    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python 3.8
      uses: actions/setup-python@v2
      with:
        python-version: 3.8
    - uses: actions/cache@v2
      with:
        path: ~/.cache/pip
        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
        restore-keys: |
          ${{ runner.os }}-pip-
    - name: Install dependencies
      run: |
        python -m pip install --upgrade pip
        pip install pytest
        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
    - name: Update odometer.csv
      run: echo `date -u +%FT%TZ`,`voc -u ${{ secrets.username }} -p ${{ secrets.password }} print odometer` >> odometer.csv
    - name: Generate volvo.html
      run: make html
    - uses: stefanzweifel/git-auto-commit-action@v4
      with:
        commit_message: Scheduled daily update
```

**Comment by staabm** [comment]
we see the same problem. a workflow which worked before errors now with the latest version

**Comment by stefanzweifel** [comment]
@morberg @staabm Thanks for reporting!

I've just released `v4.5.1` which removes the mentioned orphan feature from #95 again.
Seems I was a bit too optimistic that it wouldn't break anything. Sorry for breaking your workflows! 

I have a long standing note to finally add a test suite to this action. Will do some work over the next few days/week to make sure stuff like this isn't happening again.

**Comment by morberg** [comment]
I just tested on one of my repos and it seems to work fine again. Thanks for the swift response.
